libsignal-service/src/push_service.rs

```diff
@@ -1184,3 +1183,18 @@ pub trait PushService: MaybeSend {
         Ok(res)
     }
 }
+
+trait PhoneNumberExt {
+    fn mcc(&self) -> Option<&str>;
+    fn mnc(&self) -> Option<&str>;
+}
+
+impl PhoneNumberExt for PhoneNumber {
+    fn mcc(&self) -> Option<&str> {
+        self.carrier().and_then(|carrier| carrier.get(0..3))
+    }
+
+    fn mnc(&self) -> Option<&str> {
+        self.carrier().and_then(|carrier| carrier.get(3..))
+    }
+}
```

Comment:
I was also thinking, maybe we want some tests on this code , and if it works, we can also put it in rust-phonenumber... Some tables here might help: https://en.wikipedia.org/wiki/Mobile_country_code